### ISSUE TITLE
Possible New Admin Commands (Server chat)

### ISSUE THREAD

**Issue by CheeseJedi** [zorkind/Hellion-Rescue-Project#239]
@zorkind Some time ago Hype365 and I put together a list of additional chat commands that we thought the dedi could benefit from having - they were submitted to ZG but I don't think any of them were actually implemented in the original game. (I just found this text file today!)

It may be that some similar functionality has been added already to HRP - if not, here's the list we made as it may be of benefit to this project. 👍 

EDIT: This looks like it might be a similar (but more fleshed out) version of GTink911's issue #207 

**Possible New Admin Commands (for 0.2.8 and beyond)- by CheeseJedi and Hype365**

Some of these would benefit from server chat responses targeted at the admin (not visible by everyone)

Server Management
=================

---save		Triggers a server-side save. Server should notify once complete via chat message.

---restart <time in seconds, defaults to zero>	Triggers a server restart (with save in shutdown as usual) in the specified number of seconds, defaults to zero for immediate restart - server to send chat message notifying of this to all connected players.

---broadcast	Sends a server chat message to all connected players_as the server_ rather than as the player.

Ship/Module Management
======================

---repair	Repairs the module you are inside (or looking at), and connected modules (similar to the refill command).

---delete	Similar to the torpedo, but without the theatrics ;) Affects module you are inside (or looking at), _and_ all connected modules. Easy clean-up on aisle four.

---power [1/0]	Activates and deactivates power for the module the admin is inside (or looking at) and all connected modules/ships. Insta-power up or down.

---turrets [1/0]	Activates or deactivates turrets for the module the admin is inside (or looking at) and all connected modules/ships.

---undockall	       Causes all docking ports of the module you're inside to undock from any attached modules, not recursive.

---undockallconnected	Causes _all_ attached modules to undock from each other - dismantle an entire station.

---refreshparts	Sets the health of all dynamic objects that are parts of _systems_ up to 100% health (like a refill command but for parts in systems) in the module you are in and all connected.

---refreshall	Sets the health of all dynamic objects in any slot in the module you are in and all connected.

Security / Ownership
====================

---assign [ < Steam64ID > / < Playername > ]	Adds the specified player to the crew list - first assigned will be the 'owner'. If none specified the admin is added.

---revoke [ < Steam64ID > / < Playername > ]	Removes the specified player from the crew list. Perhaps auto promote the next player to owner if the owner was just removed? If none specified the admin is removed.

---clear	Clears the security crew list.

Cryopod Management
==================

---cryoclaim [ < Steam64ID > / < Playername > ]		Claims the looked-at (targeted control panel?) cryo for the specified player. If none specified the pod is claimed by the admin.

---cryoassign [ < Steam64ID > / < Playername > ]	Assigns (and claims if necessary) the looked-at (targeted control panel?) cryo for the specified player. If none specified the pod is assigned to the admin (becoming the spawn point)

---cryounlock	Clears the looked-at (targeted control panel?) cryo ownership information.

Inventory Management
====================

---refill (modification!)	Refill command to include electricity 'tanks' also - jetpack battery, ship/station capacitor etc. Test/admin more effectively :)

---addresource < Resource > < Amount >	Adds the specified amount of the specified resource to the container in the players hands, or a module the admin is inside, up to the maximum the container can hold. Perhaps make the amount accept a negative number also?

Player Management
=================

---listplayers	Displays a list of players: Player Name, Steam64ID, time connected(if available), ping.

---kick [ < Stem64ID > / < Playername > ]	Disconnects the specified player.

---ban [ < Stem64ID > / < Playername > ]	Adds the specified player's Steam64ID to a server-side black list (perhaps in the server config ini file, or a separate blacklist file?) STRETCH GOAL! This is a wish-list after all!

**Comment by zorkind** [comment]
Thanks a lot Cheese this helps a lot for sure :) nice to see you in here with us o/

**Comment by GTink911** [comment]
I believe the 'refill' command already fills up power. Also merging #207.

**Comment by zorkind** [comment]
Some of these commands already exists for sure in what we did in HRP but it helps anyway.
This is a shot of how the server commands are today, this is just the beginning of it.
Also keep in mind that a web interface is planned that will interface with these commands and log.
This is not available in the F2 key but there is also plans to do it, and i think the list you gave above will help a lot for sure.
![image](https://user-images.githubusercontent.com/11562921/90989637-ad502f80-e571-11ea-94da-7ff24df21780.png)

**Comment by CheeseJedi** [comment]
> I believe the 'refill' command already fills up power. 

Yep ZG did implement that shortly after the list was submitted - it was a right pain testing stuff prior, as power wasn't originally included in the refill command and required faffing about to work around it in some cases.

Sadly I don't think any of the others made it in to the original game though. Now there's a new opportunity! :)

**Comment by zorkind** [comment]
For sure :)

I don't think the --refill works on component Health tho, for example batteries  you have on  your inventory or on your helmet, wont get their charge (health) reset to 1 (total).
